tests/e2e/skew-protection.test.ts

```diff
@@ -0,0 +1,574 @@
+
+      await publishDeploy(skewProtection.siteId, skewProtection.deployA.deployID)
+
+      // poll to ensure deploy was restored before continuing
+      while (true) {
+        const response = await fetch(pollURL)
+        const text = await response.text()
+        if (text.startsWith('A')) {
+          break
+        }
+        await new Promise((resolve) => setTimeout(resolve, 50))
+      }
+
+      const result = await callback()
+
+      await publishDeploy(skewProtection.siteId, skewProtection.deployB.deployID)
+
+      // https://netlify.slack.com/archives/C098NQ4DEF6/p1758207235732189
+      await new Promise((resolve) => setTimeout(resolve, 3000))
+
+      // poll to ensure deploy was restored before continuing
+      while (true) {
+        const response = await fetch(pollURL)
+        const text = await response.text()
+        if (text.startsWith('B')) {
+          break
+        }
+        await new Promise((resolve) => setTimeout(resolve, 50))
+      }
+
+      return result
+    }
+
+    await use(fixture)
+  },
+  skewProtection: [
+    async ({}, use) => {
+      const { siteId, url } = await createSite({
+        name: `next-skew-tests-${Date.now()}`,
+      })
+
+      let onBuildStart: () => void = () => {}
+      const waitForBuildStart = new Promise<void>((resolve) => {
+        onBuildStart = () => {
+          resolve()
+        }
+      })
+
+      const deployAPromise = createE2EFixture('skew-protection', {
+        siteId,
+        useBuildbot: true,
+        onBuildStart,
+        env: {
+          NETLIFY_NEXT_SKEW_PROTECTION: 'true',
+        },
+      })
+
+      // we don't have to wait for deployA to finish completely before starting deployB, but we do have to wait a little bit
+      // to at least when build starts building, as otherwise whole deploy might be skipped and only second deploy happens
+      await waitForBuildStart
+
+      const deployBPromise = createE2EFixture('skew-protection', {
+        siteId,
+        useBuildbot: true,
+        env: {
+          NETLIFY_NEXT_SKEW_PROTECTION: 'true',
+        },
+        onPreDeploy: async (fixtureRoot) => {
+          await execaCommand(
+            `${getBuildFixtureVariantCommand('variant-b')} --apply-file-changes-only`,
+            {
+              cwd: fixtureRoot,
+            },
+          )
+        },
+      })
+
+      const [deployA, deployB] = await Promise.all([deployAPromise, deployBPromise])
+
+      const fixture = {
+        url,
+        siteId,
+        deployA,
+        deployB,
+
+        cleanup: async () => {
+          if (process.env.E2E_PERSIST) {
+            console.log(
+              `💾 Fixture and deployed site have been persisted. To clean up automatically, run tests without the 'E2E_PERSIST' environment variable.`,
+            )
+
+            return
+          }
+
+          await deployA.cleanup()
+          await deployB.cleanup()
+          await deleteSite(siteId)
```

Comment:
micro-optimization - if we're deleting the site, I don't think we need to delete both deploys first

Reply:
Individual deploy cleanups here are to handle local fixture deletion
https://github.com/opennextjs/opennextjs-netlify/blob/184b506fa366fe9a9bb9683410955691cfc9c9c6/tests/utils/create-e2e-fixture.ts#L306-L313